src/core/models/middleware/PreprocessingMiddleware.ts

```diff
@@ -0,0 +1,26 @@
+import { KintRequest } from "../KintRequest";
+import { KintResponse } from "../KintResponse";
+
+export type PreProcessingMiddleware<
+  Config extends object,
+  RequestExtension extends object = {}
+> = {
+  /**
+   *
+   * This function is called before the request is processed by the core Kint logic. It can be used to modify the request object
+   * or send a response immediately to the user.
+   *
+   * To modify the request object, return a new object which you would like to be merged with the request object and passed
+   * to the next middleware or the endpoint handler.
+   *
+   * To send a response immediately, return a KintResponse object.
+   *
```

Comment:
For the save of brevity
```suggestion
   * To send a response immediately, return a KintResponse object, this will stop the request from reaching the endpoint handler
```